browse-next/src/components/DeviceSetupDefineMask.vue

```diff
@@ -0,0 +1,823 @@
+  } else {
+    clearMask();
+  }
+};
+
+const toggleInclusionRegion = () => {
+  inclusionRegion.value = !inclusionRegion.value;
+  generateMask();
+};
+
+function removePoint(): void {
+  if (points.value.length > 1) {
+    points.value.pop();
+    drawPolygon();
+  } else {
+    points.value = [];
+  }
+  clearMask();
+  generateMask();
+  drawPolygon();
+}
+
+function addRegionSelection(): void {
+  regionsArray.value.push({ regionData: points.value });
+  points.value = [];
+  toggleCreatingRegion();
+  const canvasElement = canvas.value;
+  const context = canvasElement.getContext("2d");
+  context.clearRect(0, 0, canvasElement.width, canvasElement.height);
+  generateMask();
+}
+
+function generateMask() {
+  polygonClosedTolerance.value = devicePixelRatio.pixelRatio.value * 10;
+  if (regionsArray.value.length === 0) {
+    return;
+  }
+  const maskCanvas = canvas.value;
+  const maskContext = maskCanvas.getContext("2d");
+  maskCanvas.width = cptvFrameWidth.value * devicePixelRatio.pixelRatio.value;
+  maskCanvas.height = cptvFrameHeight.value * devicePixelRatio.pixelRatio.value;
+
+  if (inclusionRegion.value) {
+    regionsArray.value.forEach((region) => {
+      maskContext.beginPath();
+      const regionData = region.regionData;
+      maskContext.moveTo(
+        regionData[0].x * maskCanvas.width,
+        regionData[0].y * maskCanvas.height
+      );
+      for (let i = 1; i < regionData.length; i++) {
+        maskContext.lineTo(
+          regionData[i].x * maskCanvas.width,
+          regionData[i].y * maskCanvas.height
+        );
+      }
+      maskContext.closePath();
+      maskContext.fillStyle = "rgba(0, 0, 0, 0.7)";
+      maskContext.fill("evenodd");
+    });
+  } else {
+    maskContext.beginPath();
+    maskContext.moveTo(0, 0);
+    maskContext.lineTo(maskCanvas.width, 0);
+    maskContext.lineTo(maskCanvas.width, maskCanvas.height);
+    maskContext.lineTo(0, maskCanvas.height);
+    maskContext.lineTo(0, 0);
+
+    regionsArray.value.forEach((region) => {
+      const regionData = region.regionData;
+      maskContext.moveTo(
+        regionData[0].x * maskCanvas.width,
+        regionData[0].y * maskCanvas.height
+      );
+      for (let i = 1; i < regionData.length; i++) {
+        maskContext.lineTo(
+          regionData[i].x * maskCanvas.width,
+          regionData[i].y * maskCanvas.height
+        );
+      }
+      maskContext.closePath();
+    });
+
+    maskContext.fillStyle = "rgba(0, 0, 0, 0.7)";
+    maskContext.fill("evenodd");
+  }
+  drawRegionIndices();
+}
+
+const clearMask = () => {
+  const canvasElement = canvas.value;
+  const context = canvasElement.getContext("2d");
+  context.clearRect(0, 0, canvasElement.width, canvasElement.height);
+};
+
+function deleteRegion(index: number) {
+  regionsArray.value.splice(index, 1);
+  clearMask();
+  generateMask();
+}
+
+function toggleCreatingRegion(): void {
+  creatingRegion.value = !creatingRegion.value;
+  if (creatingRegionAborted.value) {
```

Comment:
`creatingRegionAborted` probably didn't need to be a reactive `ref` - it looks like it should just be an argument to the `toggleCreateRegion` function.